shiny/_template_utils.py

```diff
@@ -0,0 +1,200 @@
+    complicated the logic will diverge
+    """
+
+    component_type = questionary.select(
+        "What kind of component do you want to build?:",
+        choices=[
+            Choice(title="Input component", value="js-input"),
+            Choice(title="Output component", value="js-output"),
+            Choice(title="React component", value="js-react"),
+            back_choice,
+            cancel_choice,
+        ],
+        style=styles_for_questions,
+    ).ask()
+
+    if component_type == "back":
+        template_query()
+        return
+
+    if component_type is None:
+        sys.exit(1)
+
+    # As what the user wants the name of their component to be
+    component_name = questionary.text(
+        "What do you want to name your component?",
+        instruction="Name must be dash-delimited and all lowercase. E.g. 'my-component-name'",
+        validate=ComponentNameValidator,
+    ).ask()
+
+    if component_name is None:
+        sys.exit(1)
+
+    appdir = questionary.path(
+        "Enter destination directory:",
+        default=build_path_string(component_name),
+        only_directories=True,
+    ).ask()
+
+    if appdir is None:
+        sys.exit(1)
+
+    app_dir = copy_template_files(
+        appdir, component_type, template_subdir="package-templates"
+    )
+
+    # Print messsage saying we're building the component
+    print(f"Setting up {component_name} component package...")
+    update_component_name_in_template(app_dir, component_name)
+
+    should_install_deps = questionary.confirm(
+        "Do you want to install js dependencies now?"
+    ).ask()
+
+    success_installing_deps = True
+    if should_install_deps:
+        success_installing_deps = install_js_dependencies(app_dir)
+    else:
+        print("Skipping installing NPM deps. Run `npm install` to install them later.")
+
+    if success_installing_deps:
+        print("Successfully installed NPM dependencies.")
+    else:
+        print(
+            "Error: Failed to install NPM dependencies. You may need to install Node/npm. After doing this you can try installing manually."
+        )
+
+    print("\nNext steps:")
+    print(f"- Run `cd {app_dir}` to change into the new directory")
+    if not should_install_deps or not success_installing_deps:
+        print("- Run `npm install` to install dependencies")
+    print("- Run `npm run build` to build the component")
+    print("- Install package locally with `pip install -e .`")
+    print("- Open and run the example app in the `example-app` directory")
+
+
+def build_path_string(*path: str):
+    """
+    Build a path string that is valid for the current OS
+    """
+    # If no args are provided we should add an empty path to the list
+    # so that the result is properly formed. E.g. "./" on Unix systems
+    prefix = f".{os.path.sep}"
+    if len(path) == 0:
+        return prefix
+    return f"{prefix}{str(Path(*path))}"
```

Comment:
I think this can just be:

```suggestion
    return os.path.join(".", *path)
```


The existing behavior:

```
>>> shiny._template_utils.build_path_string()
'./'
>>> shiny._template_utils.build_path_string("a")
'./a'
>>> shiny._template_utils.build_path_string("a", "b")
'./a/b'
>>> shiny._template_utils.build_path_string("a/b", "c")
'./a/b/c'
```

The one thing I see above that's a bit inconsistent is that when there's no input, returns `./` with a trailing slash, whereas with all the other cases, it doesn't have a trailing slash.


With my suggested change:

```
>>> shiny._template_utils.build_path_string()
'.'
>>> shiny._template_utils.build_path_string("a")
'./a'
>>> shiny._template_utils.build_path_string("a", "b")
'./a/b'
>>> shiny._template_utils.build_path_string("a/b", "c")
'./a/b/c'
```

The only difference in output is the `.` vs `./`, but I think that's actually more internally consistent and will lead to more predictable behavior (e.g. if you append a path string and forget to use a leading slash, it will always result in an error, instead of only in some cases).